File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string input);
int count_words(string input);
int count_sentences(string input);

int main(void)
{
    // Initialize variables
    string input = get_string("Text: ");
    int letters = count_letters(input);
    int words = count_words(input);
    int sentences = count_sentences(input);

    // Calculate the value of L with the quantity of letters and words
    float L = 100 * (float) letters / (float) words;
    // Calculate the value of S with the quantity of sentences and words
    float S = 100 * (float) sentences / (float) words;

    // Calculates with the grade with Coleman-Liau formula
    float grade = 0.0588 * L - 0.296 * S - 15.8;

    // Check the grade and print it
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int)round(grade));
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

// Receive the input from user and return the quantity of letters in it
int count_letters(string input)
{
    int i = 0;
    int letters = 0;
    // Loop while i is lower than length of input
    while (i < strlen(input))
    {
        // Check if the caracter in input at position i is alphabetical
        if (isalpha(input[i]))
        {
            // Add 1 to letters counter
            letters++;
        }
        i++;
    }

    // Return quantity of letters
    return letters;
}

// Receive the input from user and return the quantity of words in it
int count_words(string input)
{
    int i = 0;
    // Declare variable with value 1 initialized to offset last word
    int words = 1;
    // Loop while i is lower than length of input
    while (i < strlen(input))
    {
        // Check if the caracter in input at position i is a space
        if (isspace(input[i]))
        {
            // Add 1 to words counter
            words++;
        }
        i++;
    }

    // Return the quantity of words
    return words;
}

// Receive the input from user and return the quantity of sentences in it
int count_sentences(string input)
{
    int i = 0;
    int sentences = 0;
    // Loop while i is lower than length of input
    while (i < strlen(input))
    {
        // Check if the caracter in input at position i is a dot, an exclamation mark or a question mark
        if (input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            // Add 1 to sentences counter
            sentences++;
        }
        i++;
    }
    return sentences;
}

File: psets/pset6/readability/readability.py
import cs50
import string


def main():
    # Get input from user
    text = cs50.get_string("Text: ")
    # Initialize variables that is gonna be used by counters
    letters = 0
    words = 1
    sentences = 0

    # Loop through all string provided by user
    for index in text:
        # Check if index is a type of '!, . or ?' punctuation assuming that is a final of phrase
        if index == '!' or index == '.' or index == '?':
            # Add 1 to sentences counter if true
            sentences += 1
        # If it is another type of punctuation just continue loop
        elif index in string.punctuation:
            continue
        # Check if index is a white space assuming that is a final of word
        elif index in string.whitespace:
            # Add 1 to words counter
            words += 1
        # Else add 1 to letters counters
        else:
            letters += 1

    # Calculates the average number of words by 100
    factorWord = words / 100
    # Calculates the average of Letter by average of Words
    factorLetters = letters / factorWord
    # Calculates the average of Sentences by average of words
    factorSentences = sentences / factorWord

    # Calculates the grade according to Coleman-Liau index
    grade = round((0.0588 * factorLetters) - (0.296 * factorSentences) - 15.8)

    # Check conditions and print the grade of text provided by user
    if grade < 16 and grade >= 0:
        print(f"Grade {round(grade)}")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


main()